sale_financial_risk/models/sale.py

```diff
@@ -32,3 +32,79 @@ def action_confirm(self):
                     'continue_method': 'action_confirm',
                 }).action_show()
         return super(SaleOrder, self).action_confirm()
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    amt_to_invoice = fields.Monetary(
+        string="Amount To Invoice",
+        compute='_compute_amount_to_invoice',
+        compute_sudo=True,
+        store=True
+    )
+
+    @api.depends('state',
+                 'price_reduce_taxinc',
+                 'qty_delivered',
+                 'invoice_lines',
+                 'invoice_lines.price_total',
+                 'invoice_lines.invoice_id',
+                 'invoice_lines.invoice_id.state',
+                 'invoice_lines.invoice_id.type',
+                 'invoice_lines.invoice_id.refund_invoice_ids',
+                 'invoice_lines.invoice_id.refund_invoice_ids.state',
+                 'invoice_lines.invoice_id.refund_invoice_ids.amount_total')
+    def _compute_amount_to_invoice(self):
+        """ Compute the taxed amount already invoiced from the sale order
+            line. This amount is computed as
+                SUM(inv_line.price_total) - SUM(ref_line.price_total)
+            where
+                `inv_line` is a customer invoice line linked to the SO line
+                `ref_line` is a customer credit note (refund) line linked to
+                    the SO line
+
+            Then, compute amount to invoice as:
+                total_sale_line - amount_invoiced
+            where
+                `amount_invoiced` is taxed amount previously explained
+        """
+        refund_lines_product = self.env['account.invoice.line']
+        for line in self:
+            invoice_lines = line.invoice_lines.filtered(
+                lambda l: l.invoice_id.state in ['open', 'in_payment', 'paid'])
```

Comment:
Use `{'open', 'in_payment', 'paid'}` for performance.

Reply:
Done